### DISCUSSION TITLE
Making an AUR package for Kryptor

### DISCUSSION THREAD

**Discussion by replydev** [samuel-lucas6/Kryptor#54] [Ideas]
Hi and thank you for making kryptor. It would be nice to have your software on the [Arch User Repository](https://aur.archlinux.org/), which provides a lot of user maintained packages to automate software installation / upgrade, in a similar way as Scoop or Chocolately.

My proposal is to make two packages:

- **kryptor**, which compiles from source and installs the compiled binary.
- **kryptor-bin**, which downloads the precompiled binary from GitHub Releases and installs it.

As i got some experience on maintaining AUR packages for my side project, i would be happy to maintains also the packages for Kryptor.

Tell me what you think about.

Regards

**Comment by samuel-lucas6** [comment]
That would be amazing; thank you! Is it common practice to have separate source and precompiled packages on AUR? I'd say just do one if it's easier. Another question is could the build from source be done using the Releases source code?

**Reply by replydev** [reply]
Yes it's a common practice to have separate package. 
An user could choose to do a faster installation using the bin package or the source package which is better from a security perspective.

Of course the source building would use the source tarball from releases, but some maintainers also provide a third package, for example **kryptor-git**, which compiles from the latest commit of the main branch.

I usually think that is not necessary to have a very bleeding edge package, especially for a security app like kryptor, so i would no create this last.

**Comment by replydev** [comment]
I've created the proposed packages:

- [kryptor](https://aur.archlinux.org/packages/kryptor)
- [kryptor-bin](https://aur.archlinux.org/packages/kryptor-bin)

@samuel-lucas6 when you can, can you confirm me that the build command used in in kryptor [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=kryptor) is right? As i see, your compiled binaries size is around 35MB, but mine is 64MB, so probably i'm missing something.

Regards.

**Reply by samuel-lucas6** [reply]
I'll go ahead and link them on the website in a minute. Looks like you're missing `-p:PublishTrimmed=true` and `-p:IncludeNativeLibrariesForSelfExtract=true`. The full command I use is [here](https://www.kryptor.co.uk/installation#using-the-.net-cli).

**Reply by replydev** [reply]
Thank you, i've added the missing flags to the build command. Seems fine now.

**Reply by samuel-lucas6** [reply]
Thank you again as well. I've linked the packages and your GitHub on the website.
